src/metadataGeneration/tsoa.ts

```diff
@@ -82,7 +92,7 @@ export namespace Tsoa {
     elementType: Type;
   }
 
-  export interface ReferenceType extends Type {
+  export interface ReferenceType extends Type, RefTypeMinimal {
     description?: string;
     dataType: 'refObject' | 'refEnum';
```

Comment:
why not have this as `RefTypeLiteral`?

Reply:
Are you asking if `RefTypeMinimal` should be called `RefTypeLiteral`? `RefTypeMinimal` isn't a string literal, it's an object. So I think calling it `RefTypeLiteral` would be confusing. 

```ts
  export interface RefTypeMinimal {
    dataType: RefTypeLiteral;
  }
```

But I am open to changing the name if you have other ideas for `RefTypeMinimal`.

Reply:
Oh, I think I understand why `dataType` isn't `RefTypeLiteral`. I can make that change.